### ISSUE TITLE
Би било хубаво ако се правят checkpoints по време на обучение

### ISSUE THREAD

**Issue by IliyaKarkamov** [MariyaZhelyazkova/NN-UniversityProject#3]
ще е хубаво да се правят checkpoints защото иначе при спиране на тока по средата на обучение губим наученото до сега.

**Comment by alefremov** [comment]
Да, може на всяка n-та стъпка да save-аш резултата в един if...
Това означава, че ще трябва допълнителна логика при пускане на кода отново - да продължи от епохата до която е стигнал.
Многократното записване ще забави обучението. Ако това ни притеснява може да изкараш един параметър, с който да кажеш дали да се записват междинно резултати.
